src/manifold/src/edge_op.cpp

```diff
@@ -325,12 +323,10 @@ void Manifold::Impl::DedupeEdge(const int edge) {
     vertPos_.push_back(vertPos_[endVert]);
     if (vertNormal_.size() > 0) vertNormal_.push_back(vertNormal_[endVert]);
 
-    do {
+    ForVert(current, [this, newVert](int current) {
       halfedge_[current].endVert = newVert;
-      current = NextHalfedge(current);
-      halfedge_[current].startVert = newVert;
-      current = halfedge_[current].pairedHalfedge;
-    } while (current != edge);
+      halfedge_[halfedge_[current].pairedHalfedge].startVert = newVert;
```

Comment:
This should be `halfedge_[NextHalfedge(current)].startVert = newVert`?

Reply:
Not quite, but that was definitely the bug. Turns out I was orbiting the wrong end of the edge.